walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,156 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> slots = new ArrayList<TimeRange>();
+
+    ArrayList<Event> eventsArr = sortEventsByStart(events);
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return slots;
+    }
+
+    slots.add(TimeRange.WHOLE_DAY);
+
+    if (!request.getOptionalAttendees().isEmpty()) {
+      ArrayList<TimeRange> optionalSlots = findTimeSlots(eventsArr, request, slots, true);
```

Comment:
When a method has multiple parameters like this, it can be hard to keep track of what each param means especially if it has a value like true or false.  In this case, we can leave a comment to better document what this arg means:
```suggestion
      ArrayList<TimeRange> optionalSlots = findTimeSlots(eventsArr, request, slots, /* hasOptionalAttendees= */ true);
```

Reply:
And for this parameter, if I'm understanding its meaning correctly, it's not exactly about whether the request has optional attendees, but whether to only return time slots that the optional attendees can attend. So `treatOptionalAttendeesAsRequired` or `mustIncludeOptionalAttendees` or something would be better IMO.